docs/API.md

```diff
@@ -483,3 +460,46 @@ The React components provided by React Intl allow for a declarative, idiomatic-R
 **See:** The [Components][components] page.
 
 [components]: Components.md
+
+## Advanced Usage
+
+For scenarios where performance is needed, you can pass in `messages` that contains `intl-messageformat`'s `AST` as values to `IntlProvider` to save compilation time when formatting messages. This is especially useful for:
+
+1. Server-side rendering where you can cache the AST and don't have to pay compilation costs multiple time.
+2. Desktop apps using Electron or CEF where you can preload/precompile things in advanced before runtime.
+
+Example:
+
+```tsx
+import parser from 'intl-messageformat-parser';
+import * as ReactDOM from 'react-dom';
+const messages = {
+  ast_simple: parser.parse('hello world'),
+  ast_var: parser.parse('hello world, {name}'),
+};
+
+ReactDOM.render(
+  <IntlProvider messages={messages}>
+    <FormattedMessage id="ast_simple" />
+  </IntlProvider>
+); // will render `hello world`
+```
+
+If you're guaranteed to always provide `react-intl`, you can use our core bundle which is significantly smaller in size by swapping out `react-intl` `import`:
```

Comment:
What does this mean, guaranteed to provide react-intl? In what scenarios wouldn’t you?